pkg/ebpf/map.go

```diff
@@ -70,13 +71,42 @@ func (m *Map) OpenOrCreate() error {
 	mapType := bpf.GetMapType(bpf.MapType(m.spec.Type))
 	m.spec.Flags = m.spec.Flags | bpf.GetPreAllocateMapFlags(mapType)
 
+	path := bpf.MapPath(m.spec.Name)
```

Comment:
I think we're still missing one other difference between the implementations: `pkg/bpf/bpf_linux.go:OpenOrCreateMap()` will attempt to create the parent directory here before pinning; from a brief search through the ebpf library I didn't see the same logic.

Might not be too relevant for this PR depending on the order of map creation but we'll want to fix this up at some point before we fully switch libraries over.

Reply:
Yep, I initially ignored it as `cilium_metrics` is not the first pinned map opened by the agent, but since we are on it I just added the logic to `ebpf.OpenOrCreate()`